### PR TITLE
[he_IL] Add lorem and text

### PR DESCRIPTION
**Pull request by MordiSacks**
### What is the reason for this PR?

I have add Lorem and Text for he_IL

- [x] A new feature
- [ ] Fixed an issue (resolve #ID)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Summary of changes

Lorem.php and Text.php have bean added to Providers/he_IL (along with appropriate tests)
`strlen` in `Faker\Provider\Lorem:199` has been changed to `mb_strlen` to allow proper sizing in he_IL lorem 

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

**Comment by pimjansen** [comment]
Build fails

File: src/Faker/Provider/Lorem.php [hunk 1]
```diff
@@ -196,7 +196,7 @@ public static function text($maxNbChars = 200)
                 $word = ($size ? ' ' : '') . static::$type();
                 $text[] = $word;
 
-                $size += strlen($word);
+                $size += mb_strlen($word);
```
Comment (pimjansen) [inline src/Faker/Provider/Lorem.php:199]: This is a breaking change where you require an additional php extension for this.

We can do this with the polyfil but not like this

**Review by pimjansen — CHANGES_REQUESTED** [review]

File: src/Faker/Provider/Lorem.php (hunk 1 above)
Comment (MordiSacks) [reply src/Faker/Provider/Lorem.php:199]: I understand and I agree, this should not  break.  
I don't understand how the polyfil can help us as it uses `iconv` and we don't requite `ext-iconv`.  
maybe something like this?

```php
    /**
     * @param string      $string
     * @param string|null $encoding
     *
     * @return int
     */
    public static function strlen($string, $encoding = null)
    {
        if(method_exists(get_called_class(), 'provider_strlen')){
            return call_user_func([get_called_class(), 'provider_strlen'], $string, $encoding);
        }

        return strlen($string);
    }
```
Then Lorem would call 
```php
$size += static::strlen($word);
```
And my Lorem will do
```php
    public static function provider_strlen($string, $encoding = null)
    {
        return mb_strlen($string, $encoding);
    }
```

Comment (MordiSacks) [reply src/Faker/Provider/Lorem.php:199]: Looking at this now, I have no idea why I reached out for this, I think simple inheritance would work fine here

**Pull request closed without merging**